src/routes/safe/components/Transactions/helpers/TxModalWrapper/index.tsx

```diff
@@ -45,6 +48,19 @@ type Props = {
 const Container = styled.div`
   padding: 0 ${lg} ${md};
 `
+export const isApproveAndExecute = (
+  threshold: number,
+  txConfirmations: number,
+  txType?: string,
+  preApprovingOwner?: string,
+): boolean => {
+  if (txConfirmations === threshold) return false
+  if (!preApprovingOwner) return false
+  return txConfirmations + 1 === threshold || isSpendingLimit(txType)
+}
+
+export const isTxCreation = (txConfirmations: number, txType?: string): boolean =>
+  txConfirmations === 0 && !isSpendingLimit(txType)
```

Comment:
This name seemed a bit off for the returning condition and I was investigating the possible values of `txType`.
I think we can narrow it down from `string` to `'multiSig' | 'spendingLimit'`.

My suggestion is that we modify first in the `<ReviewSendFundsTx />` component props and then the subsequent prop drilling.

If you think it is out of scope I suggest we rename the function to `isMultisigCreation`

Reply:
Very good suggestion. It looks like we hardcode those values everywhere so far but using a restrictive type would be a lot better. I will create a separate task for it as it involves a lot of changes outside of the scope of this ticket.

Reply:
If you are creating a new task for it, fine. The rename would be useful nevertheless

Reply:
Don't we use `isCreation` in the Safe as a flag for the Safe creation?

Reply:
Issue for the refactor: https://github.com/gnosis/safe-react/issues/3518